gateway-ha/src/main/java/io/trino/gateway/baseapp/AppModule.java

```diff
@@ -46,11 +46,19 @@ protected int getApplicationPort()
                         .map(SimpleServerFactory::getConnector);
 
         return connectors
-                .filter(connector -> connector.getClass().isAssignableFrom(HttpConnectorFactory.class))
-                .map(connector -> (HttpConnectorFactory) connector)
-                .mapToInt(HttpConnectorFactory::getPort)
-                .findFirst()
-                .orElseThrow(IllegalStateException::new);
+              .filter(connector -> connector instanceof HttpConnectorFactory || connector instanceof HttpsConnectorFactory)
+              .map(connector -> {
+                  if (connector instanceof HttpConnectorFactory) {
+                      return ((HttpConnectorFactory) connector).getPort();
+                  } else if (connector instanceof HttpsConnectorFactory) {
+                      return ((HttpsConnectorFactory) connector).getPort();
+                  } else {
+                      throw new IllegalStateException("Unknown connector type");
+                  }
```

Comment:
The format is wrong. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#code-style